[ADD] payment_providers: Xendit (#6045)
odoo/documentation: content/applications/finance/payment_providers/xendit.rst

======
Xendit
======

`Xendit <https://www.xendit.co>`_ is an Indonesian-based payment solution provider that covers
several Southeast Asian countries. It allows businesses to accept credit cards as well as several local
payment methods.

.. _payment_providers/xendit/configure_dashboard:

Configuration on the Xendit Dashboard
=====================================

#. Create a Xendit account if necessary and log in to the `Xendit Dashboard
   <https://dashboard.xendit.co>`_.
#. Check your account mode in the top left corner of the page. Use the :guilabel:`Test Mode` to try the integration
   without charging your customers. Switch to :guilabel:`Live Mode` once you are ready to accept
   payments.
#. Navigate to :menuselection:`Configuration: Settings` in the left part of the application page. In the :guilabel:`Developers` section,
   click `API Keys <https://dashboard.xendit.co/settings/developers#api-keys>`_. In tis page, you will setup for both :guilabel:`API key`

>    and :guilabel:`webhook token`.

Reviewer: I don't think the second part of the sentence is necessary. I think it's fine if users follow the rest of the procedure and find out what they need to when they get to it?
If you really want to keep it, then I would change it to:

This page allows you to set up both the API key and webhook token. 
(no guilabel, but you could use bold if you want to)

> #. Click :guilabel:`Generate Secret Key`, and in the popup box, enter any :guilabel:`API key name`, select :guilabel:`Write` for
>    the :guilabel:`Money-in Products` permission and :guilabel:`None` for all other permissions
>    to keep your data secured, then click :guilabel:`Generate key`.

Reviewer: This sentence seems a little long so I would rephrase it as follows:
```suggestion
#. Click :guilabel:`Generate Secret Key`. In the popup box, enter an :guilabel:`API key name`, select :guilabel:`Write` for  the :guilabel:`Money-in Products` permission and :guilabel:`None` for all other permissions, then click :guilabel:`Generate key`.
```

#. Confirm your password to display your API key. Copy or download the key and **save
   this information securely for later**. This is the only time the API key can be viewed or downloaded.

> #. When completed, scroll down the page to the `Webhooks <https://dashboard.xendit.co/settings/developers#webhooks>`_ section.

Reviewer: ```suggestion
#. Once completed, scroll down the page to the `Webhooks <https://dashboard.xendit.co/settings/developers#webhooks>`_ section to generate the webhook token.
```
(added the last part since i removed it from the next bullet point and i thought it would work well here)

> #. Under :guilabel:`Webhook verification token`, click :guilabel:`View Webhook Verification Token` to generate the token.
> #. Confirm your password to display the token and save it for later.

Reviewer: I think these two can be merged:
```suggestion
#. Under :guilabel:`Webhook verification token`, click :guilabel:`View Webhook Verification Token`, then confirm your password to display the token. Save it for later.
```

#. In the :guilabel:`Webhook URL` section, enter your Odoo database URL (e.g.,
   `https://example.odoo.com`) in the field :guilabel:`Invoices paid` and click the
   :guilabel:`Test and save` button next to it.

> 

Reviewer: there a 2 empty lines here instead of just one

Configuration on Odoo
=====================

#. :ref:`Navigate to the payment provider Xendit <payment_providers/add_new>` and change its state
   to :guilabel:`Enabled`.

> #. Fill in the :guilabel:`Secret Key` and :guilabel:`Webhook Token` fields with the
>    information saved from the previous steps :ref:`payment_providers/xendit/configure_dashboard`.

Reviewer: ```suggestion
#. Fill in the :guilabel:`Secret Key` and :guilabel:`Webhook Token` fields with the
   information saved at the step :ref:`payment_providers/xendit/configure_dashboard`.
```
(nitpicking, this is just to have the same as the other (recently updated) payment provider docs)

#. In the popup box, enter any :guilabel:`API key name`, select :guilabel:`Write` for
   the :guilabel:`Money-in Products` permission and :guilabel:`None` for all other permissions
   to keep your data secured, and click :guilabel:`Generate key`.
#. After confirming your password, your API key is displayed. Copy or download the key and **save
   this information securely**. It is the only time the API key can be viewed.

Webhook token and URL
---------------------

#. Go to `Configuration: Settings --> Developers: Webhooks
   <https://dashboard.xendit.co/settings/developers#webhooks>`_. Under :guilabel:`Webhook
   Verification Token`, click :guilabel:`View Webhook Verification Token` to generate it.
#. After confirming your password, your token is displayed. Save this information securely.
#. In the :guilabel:`Webhook URL` section, enter your Odoo database URL (e.g.,
   `https://example.odoo.com`) and click :guilabel:`Test and save` for each of the following fields:

   - :guilabel:`Invoices: Invoices paid`

Configuration on Odoo
=====================

#. :ref:`Navigate to the payment provider Xendit <payment_providers/add_new>` and change its state
   to :guilabel:`Enabled`.

> #. Fill in the :guilabel:`Xendit API Key` and :guilabel:`Xendit Callback Token` fields with the

Reviewer: From what I can see in the dev PR, but feel free to correct this if I'm wrong:

```suggestion
#. Fill in the :guilabel:`Secret Key` and :guilabel:`Webhook Token` fields with the
```

>    information saved from the previous steps.

Reviewer: I would add a ref here linking to the first h2 (xendit dashboard config), like we have for other payment providers

> #. Configure the payment methods you want to activate on the configuration tab.

Reviewer: ```suggestion
#. Configure the rest of the options to your liking.
```
since it's not just the payment methods :)
